thread.c

```diff
@@ -4254,6 +4254,27 @@ rb_thread_fd_select(int max, rb_fdset_t * read, rb_fdset_t * write, rb_fdset_t *
     return (int)rb_ensure(do_select, (VALUE)&set, select_set_free, (VALUE)&set);
 }
 
+#ifdef RUBY_THREAD_PTHREAD_H
+
+static bool
+thread_sched_wait_events_timeval(int fd, int events, struct timeval *timeout)
```

Comment:
let's receive `th` as a function parameter.

Reply:
also `if (!th_has_dedicated_nt(th))` can be located at first.

```C
thread_sched_wait_events_timeval(th, ...)
{
#ifdef RUBY_THREAD_PTHREAD_H
  if (!th->nt->dedicated) {
    ...
  }
#endif // RUBY_THREAD_PTHREAD_H
}
```

and caller can be simpler like:

```C
  if (thread_sched_wait_events_timeval()) { return ...; // timeouit }
```

Reply:
Great suggestions! Much cleaner now - you can review the specific change made here https://github.com/ruby/ruby/pull/9178/commits/5543d5c6c6d80186ee5a87fa8dee59a28bb3d315

Reply:
Hmmm, since adding this commit I get the following error, exclusively on the travis CI build: https://app.travis-ci.com/github/ruby/ruby/builds/267899400

do you know what that would be @ko1 ?